.github/workflows/build.yaml

```diff
@@ -5,45 +5,13 @@ on:
     branches: [ main ]
   schedule:
     - cron: '12 0 * * *'
-defaults:
-  run:
-    shell: bash
+  workflow_dispatch:
 
 jobs:
   build_and_test:
-    name: Build and test
-    runs-on: ubuntu-latest
-    container:
-      image: rostooling/setup-ros-docker:ubuntu-jammy-latest
-    steps:
-      - name: deps
-        uses: ros-tooling/setup-ros@v0.3
-        with:
-          required-ros-distributions: humble
-      - name: install_clang
-        run: sudo apt update && sudo apt install -y clang clang-tools lld
-      - name: build
-        uses: ros-tooling/action-ros-ci@v0.2
-        env:
-          CC: clang
-          CXX: clang++
-        with:
-          target-ros2-distro: humble
-          # build all packages listed in the meta package
-          package-name: |
-            rmf_battery
-          vcs-repo-file-url: |
-            https://raw.githubusercontent.com/open-rmf/rmf/main/rmf.repos
-          colcon-defaults: |
-            {
-              "build": {
-                "mixin": ["coverage-gcc","lld"]
-              }
-            }
-          colcon-mixin-repository: https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v1
-        with:
-          files: ros_ws/lcov/total_coverage.info
-          flags: tests
-          name: lean_and_mean_codecov_bot
+    name: rmf_battery
+    uses: open-rmf/rmf_ci_templates/.github/workflows/reusable_build.yaml@main
```

Comment:
Just to check, what does the `@main` signify? Does CI only run on the `main` branch or does the action checkout to the relevant branch?

Reply:
The workflow used is the one of on the main branch of this repo: open-rmf/rmf_ci_templates/.github/workflows/reusable_build.yaml It can be useful if we need minor tweaks for some ROS distributions os specific situations but I think we should try to keep it as centralized as possible.

Reply:
Ah okay it uses the `main` version of that template. Thanks for the clarification.